jetty-ee10/jetty-ee10-tests/jetty-ee10-test-client-transports/src/test/java/org/eclipse/jetty/ee10/test/client/transport/ServerTimeoutsTest.java

```diff
@@ -463,7 +462,8 @@ protected void service(HttpServletRequest request, HttpServletResponse response)
                     @Override
                     public void onDataAvailable() throws IOException
                     {
-                        assertEquals(0, input.read());
+                        while (input.isReady())
+                            assertEquals(0, input.read());
```

Comment:
Why this? The client only sends 1 byte, we have been called in `onDataAvailable()` so when we read we must read 1 byte, as the previous code was doing.

Reply:
I think the idea was to check that both an `isReady` and a `read` could be done.    So it probably could have been done with an `if` rather than a `while`, but the `while` is the more natural style.

However, we should probably also have a test of a `read` without a prior `isReady` call (as it was before... we should test both).

Reply:
Let me add such test.